types/compute-gcd/index.d.ts

```diff
@@ -0,0 +1,15 @@
+// Type definitions for compute-gcd 1.2
+// Project: https://github.com/compute-io/gcd
+// Definitions by: Sean S. LeBlanc <https://github.com/seleb>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+type Accessor<T> = (item: T) => number;
+
+declare function gcd(firstValue: number, secondValue: number, ...values: ReadonlyArray<number>): number;
+declare function gcd(values: [number, number, ...ReadonlyArray<number>]): number;
+declare function gcd<T>(values: [T, T, ...ReadonlyArray<T>], accessor: Accessor<T>): number;
+// If provided an array with a length less than 2 or a single integer argument, the function returns `null`.
+declare function gcd(singleValue: number | [number]): null;
+declare function gcd<T>(singleValueArray: [T], accessor: Accessor<T>): null;
```

Comment:
You need (in my opinion) a fallback overload for arrays of unknown length. This is currently an error, which seems tedious:

```ts
declare let someNumbers: number[];
// $ExpectType number | null
gcd(someNumbers); // but actually it’s an error :(
```